aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -82,16 +87,17 @@ static void binaryOpTensor(const Tensor& self,
     return;
   }
 
-  Tensor output = output_;
+  Tensor output;
   bool needsCopyToOutput = false;
 
-  if (!output_.is_contiguous()) {
-    output = output_.contiguous();
-    needsCopyToOutput = true;
-    // else, determine if this is an in-place operation on a view output
-  } else if (output_.is_view() && (self.is_alias_of(output_) || other.is_alias_of(output_))) {
-    output = at::empty(output_.sizes(), output_.scalar_type(), c10::nullopt, kMPS, c10::nullopt, c10::nullopt);
-    needsCopyToOutput = true;
+  // determine if this is an in-place operation
+  if (self.is_alias_of(output_) || other.is_alias_of(output_)) {
+    if (output_.storage_offset() || !output_.is_contiguous()) {
+      output = at::empty(output_.sizes(), output_.scalar_type(), c10::nullopt, kMPS, c10::nullopt, c10::nullopt);
+      needsCopyToOutput = true;
+    }
+  } else if (!output_.is_contiguous()) {
+    output_.unsafeGetTensorImpl()->empty_tensor_restride(MemoryFormat::Contiguous);
```

Comment:
This is a no-op, isn't it?

Reply:
No, it's an optimization. Instead of performing a gather in the Placeholder on the non-contiguous tensor, we make the empty tensor contiguous with this trick. We did the same in several ops too:
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/mps/operations/Pooling.mm#L126